bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/internal/AsyncReadyCallback.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Patrick Ziegler and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.internal;
+
+import org.eclipse.swt.widgets.*;
+
+/**
+ * This class implements the GIO AsyncReadyCallback type and is used to
+ * transform an asynchronous {@code async} and synchronous {@code await}
+ * operation into a single synchronous {@code run} operation.
+ */
+public abstract class AsyncReadyCallback {
```

Comment:
Maybe make this an interface and put the run method into SyncDialogUtils (or make it default implemented)

Reply:
To be honest, I'd really like to get rid of the SyncDialogUtils class, which is why I didn't move the run method there. But the best way can also be decided at a later stage, I suppose...
In any case, this class is now an interface and the old run() method in SyncDialogUtils has been replaced.